src/ctelnet.cpp

```diff
@@ -2788,6 +2789,8 @@ void cTelnet::processSocketData(char* in_buffer, int amount)
                     // IAC SB COMPRESS WILL SE for MCCP v1 (unterminated invalid telnet sequence)
                     // IAC SB COMPRESS2 IAC SE for MCCP v2
                     if ((mMCCP_version_1 || mMCCP_version_2) && (!mNeedDecompression)) {
+                        // TODO this code looks ahead instead of using the state machine.
+                        // This is not a good idea.
```

Comment:
Perhaps it is required in order to initiate the decompression at exactly the right point in time - getting that wrong WILL break things... :man_shrugging:

Reply:
The problem is that while this code checks that the complete subrequest is in the buffer, it does nothing whatsoever if it is not, i.e. if the compression start message is ever split into two TCP packets (which conceivably might happen) it would be disregarded entirely.